app/src/main/java/com/example/scoutconcordia/Activities/MapsActivity.java

```diff
@@ -353,56 +387,152 @@ public void initializeSearchBar(){
-
+                 startingPoint = startingLocation.getText().toString();
+                 if(!enableShuttle(dialogView, locationsInDifferentCampuses(startingPoint, destination))){
+                     setModeToWalkIfShuttleSelected();
+                 }
             }
         });
-        builder.create().show();
+        return builder.create();
+    }
+
+    private boolean locationsInDifferentCampuses(String startingPoint, String destination){
+        LatLng origin = locationMap.get(startingPoint);
+        LatLng dest = locationMap.get(destination);
+
+        float[] results = new float[1];
+        Location.distanceBetween(origin.latitude, origin.longitude, dest.latitude, dest.longitude, results);
+        return results[0] > 2000.0;
+    }
+
+    private boolean enableShuttle(View v, boolean enabled){
+        BottomNavigationItemView shuttleModeItem = v.findViewById(R.id.shuttle);
+        shuttleModeItem.setEnabled(enabled);
+        shuttleAvailable = enabled;
+        return enabled;
     }
 
-    // get directions button clicked in dialog for getting directions (in onFindYourWayButtonClick)
     public void onGetDirectionsClick(View v){
+        if(useMyLocation) {
+            setOriginDialog.dismiss();
+            if (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED) {
+                getDirectionsFromCurrentLocation();
+            }
+            else{
+                requestPermissions(ACCESS_FINE_LOCATION_DRAW_PATH);
+            }
+        }
+        else { //starting point not current location
+
+            LatLng origin = null;
+            // if place is not in locationMap, exception will be thrown
+            // No need to check for destination as setOrigin only opens when one of the given options is selected
+            try{
+               origin = locationMap.get(startingPoint);
+            }
+            catch(Throwable t){
+                Log.d(TAG, ""+t.getMessage());
+                Toast.makeText(this, "Invalid starting location", Toast.LENGTH_LONG).show();
+            }
+
+            if(startingPoint != null) {
+                setOriginDialog.dismiss();
+                drawDirectionsPath(origin, locationMap.get(destination));
+            }
+        }
+    }
+
+    public void useMyLocationButtonClicked(View v){
+        RadioButton btn = (RadioButton)v.findViewById(R.id.useMyLocationButton);
+        AutoCompleteTextView startingLocation = ((ViewGroup)v.getParent().getParent()).findViewById(R.id.starting_location);
+
+        setModeToWalkIfShuttleSelected();
+        enableShuttle((ViewGroup)v.getParent().getParent(), false);
+        btn.setChecked(!useMyLocation);
+        useMyLocation = !useMyLocation;
     }
 
     // walking option selected in dialog for getting directions (in onFindYourWayButtonClick)
     public void onWalkingSelected(MenuItem m){
         m.setChecked(true);
+        travelMode = 1;
```

Comment:
Wouldn't it be safer to set this as a constant?

Reply:
It's not a constant as value needs to be changed depending on travel mode the user chooses.